### PR TITLE
Fix Helm upgrade regression tests

### PR DESCRIPTION
**Pull request by marcogschmidt**
# Description
This PR brings back the regression tests that test upgrades via `helm upgrade`. Additional changes include:
- fixed `glooctl check` so that it skips the `RateLimitConfig` check for versions that are < `1.5.0-beta8` (or < `1.5.0-beta3` for Gloo Enterprise)
- started a `1.5` upgrade guide to explain the upgrade process.

# Context

The addition of the new `RateLimitConfig` CRD caused issues in the Helm upgrade tests, as `helm upgrade` completely ignores CRDs defined in the chart. See the linked #3359 issue for more details.

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo [contribution guide](https://docs.solo.io/gloo/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/3359

### CODE REVIEW

File: docs/content/operations/upgrading/1.5.0.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+```shell script
+helm repo update
+helm fetch glooe/gloo-ee --version 1.5.0-beta3 --untar
+kubectl apply -f gloo-ee/charts/gloo/crds/ratelimit_config.yaml
+```
+{{% /tab %}}
+{{< /tabs >}}
+
+
+You can verify that the new CRD has been successfully applied by running thw following command:
```
Comment (kdorosh) [inline docs/content/operations/upgrading/1.5.0.md:82]: typo,
```suggestion
You can verify that the new CRD has been successfully applied by running the following command:
```

File: projects/gloo/cli/pkg/cmd/check/root.go [hunk 2]
```diff
@@ -548,3 +579,76 @@ func checkConnection(ns string) error {
...
+			continue
+		}
+		return false
+	}
+}
+
+// Returns true only if all the instances of Gloo running in the given namespaces have a version
+// that is less than RateLimitCrdReleaseVersion. In cases where a comparison is not possible,
+// we err on the side of caution and return false.
+func allGloosArePreRateLimitConfig(namespaces []string) (bool, error) {
```
Comment (kdorosh) [inline projects/gloo/cli/pkg/cmd/check/root.go:606]: not a huge fan of all this version-checking logic for deciding if we expect to find ratelimit CRDs.

i think it's a fair assumption that if ratelimit CRDs don't exist on the server, then we can return healthy (as we cannot find any unhealthy ratelimit CRDs).

in the rare event an installation failed to install a CRD, the error would have been reported there. i think `glooctl check` owning that check is a bit out of scope

Comment (marcogschmidt) [reply projects/gloo/cli/pkg/cmd/check/root.go:606]: Yeah, that makes sense. If the CRD is required and somehow was not installed, we would not even get to this point, as the gloo pod will not start and we check for that before checking the resources. Happy to save us some extra code, will remove.

File: projects/gloo/cli/pkg/cmd/check/root.go [hunk 3]
```diff
@@ -347,7 +356,18 @@ func checkRateLimitConfigs(namespaces []string) ([]string, bool, error) {
 	fmt.Printf("Checking rate limit configs... ")
 	var knownConfigs []string
 	for _, ns := range namespaces {
-		configs, err := helpers.MustNamespacedRateLimitConfigClient(ns).List(ns, clients.ListOpts{})
+
+		rlcClient, err := helpers.RateLimitConfigClient([]string{ns})
+		if err != nil {
+			if isCrdNotFoundErr(err) {
+				// Just warn. If the CRD is required, the check would have failed on the crashing pod.
```
Comment (kdorosh) [inline projects/gloo/cli/pkg/cmd/check/root.go:363]: the crashing gloo pod, let's be more specific here

**Pull request merged**